src/content/protocols/kodiak-v3.md

```diff
@@ -0,0 +1,202 @@
+---
+protocol: "Kodiak Finance V3"
+website: "https://www.kodiak.finance/"
+x: "https://x.com/KodiakFi"
+github: ["https://github.com/Kodiak-Finance"]
+defillama_slug: ["kodiak-v3"](https://defillama.com/protocol/kodiak-v3)
+chain: "Berachain"
+stage: 0
+reasons: []
+risks: ["L", "M", "H", "H", "H"]
+author: ["gnart33"]
+submission_date: "2025-05-07"
+publish_date: "2025-05-07"
+update_date: "2025-05-07"
+---
+
+# Summary
+Kodiak Finance V3 is a comprehensive liquidity hub on Berachain that combines multiple DeFi components into a unified platform. At its core, it offers both Uniswap V2-style (constant product) and V3-style (concentrated liquidity) AMMs, with a protocol fee of 35% from launch. 
+- "Kodiak Islands" system provides ERC20-wrapped V3 positions, where users can provide liquidity in two variants: 
+  - Managed Islands, which are protocol-run with automatic rebalancing and a 10% fee, 
+  - Unmanaged Islands that allow users to deploy their own positions without rebalancing.
+- Users can stake their LP tokens in the RewardVault contract to earn additional rewards from the protocol's incentive layer.
+- A distinctive feature of Kodiak is its integrated incentive layer and no-code token deployer factory called "Panda Factory." (which will deploy a PandaToken). This system enables token launches on Berachain using bonding curves with several features: a 1B total supply model, zero initial liquidity requirement, configurable price ranges, and support for base tokens like BERA and HONEY. The Panda Factory includes an optional same-block buy feature for deployers (up to 10% of supply) and implements a graduation mechanism where tokens transition from the bonding curve to the Kodiak DEX once 80-90% of the supply is purchased.
+- The protocol's architecture is built on immutable contracts rather than using proxy upgradeable contracts, with protocol parameters managed through multi-sig wallets.
+
+# Ratings
+
+## Chain
+
+Berachain Mainnet, an EVM-Identical Layer 1 blockchain utilizing Proof-of-Liquidity 
+
+> Chain score: Low
+
+## Upgradeability
+
+The multiple multi-sig wallets control the upgradeability/implementation/parameter changes of the protocol, including: setDefaultFeeProtocol, enableFeeAmount, setIslandFee (20% max), setTreasury. None of them show possiblity of theft or loss of user funds or loss of unclaimed yield, but malicious actions could lead to unexpected behavior of the system.
+
+> Upgradeability score: Medium
+
+## Autonomy
+
+The protocol's autonomy is controlled by three multi-sig wallets with low thresholds:
+
+1. Safe Wallet 1 (2/4): Controls UniswapV3Factory and FarmFactory
+2. Safe Wallet 2 (2/3): Controls KodiakIslandFactory
+3. Safe Wallet 3 (2/3): Controls PandaFactory
+4. Safe Wallet 4 (3/5): Controls RewardVaultFactory
+
+These wallets can immediately execute all critical functions without any timelock or governance process.
+
```

Comment:
@newgnart mentioned in another comment, that multisigs that are permission owners are not part of dependency/autonomy -> only oracles, bridges, rng ...